### DISCUSSION TITLE
An option to make shop packs unable to be discounted!

### DISCUSSION THREAD

**Discussion by elusive-ttv** [gameserverapp/Platform#494] [Suggestions]
I am returning to GSA after a while away, and I LOVE the new changes!

With that said, I would like to apply discounts by custom role. We have shop packs to convert tokens into the token gems to promote trading, and our problem currently is that tokens can just be duped by people with custom roles that include discounts. Ideally we would like to be able to turn off the ability to discount, maybe as an "Allow Discounts" checkbox in the settings of the pack.

**Comment by mvaessen** [comment]
hi, for each discount you can select which shop packs & supporter tiers it can be applied too. By simply not selecting the shop pack you dont want to sell on the discount, you should be able to achieve that.

Let me know if you still think this is a good suggestion or whether it's redundant based on this info.

PS: and welcome back!

**Reply by elusive-ttv** [reply]
I tried that, but with the custom roles it doesn't have that option. The discounts tab works great for people who don't have any custom roles with discounts, but the custom roles seem to override the discount settings and offer discounts on unselected packs anyways.

**Reply by mvaessen** [reply]
Ah, i understand now.

For future reference (note 2 self), this is the result of this setting: 
![Screenshot 2023-08-07 at 14 50 47](https://github.com/gameserverapp/Platform/assets/13402423/c32eb884-8dac-4868-875b-b553f89a1256)

**Reply by elusive-ttv** [reply]
Yes exactly that setting! It's wonderful! There are just a few packs I want to disable all discounts on, so players cannot make free tokens.

**Reply by DieselEffect** [reply]
I came here to say that I would absolutely love this function as well. I am trying to allow players to purchase token gems in-game with their tokens so they can use them with other players, but the people getting discounts would be able to exploit it.

**Reply by DieselEffect** [reply]
Actually, I think I have a way to do this without action from GSA:

- Remove all discounts in "Custom Roles" (make them 0)
- Go to the actual Discounts section and make one that has no end-date.
- Ensure the discount is role-specific
- Check every shop pack you want subjected to the discount, leave the ones you don't unchecked.
- Save

If anyone has any added advice here lemme know but I have this working as intended in my shop.

**Comment by mvaessen** [comment]
Created dev item #966

**Comment by mvaessen** [comment]
Thanks for the suggestion, this is now available. 
![Screenshot 2024-11-19 at 11 46 46](https://github.com/user-attachments/assets/869ba2a0-64bf-40c7-a363-f8db115c8012)
